python/plugins/processing/algs/qgis/TilesXYZ.py

```diff
@@ -180,102 +184,132 @@ def prepareAlgorithm(self, parameters, context, feedback):
         self.layers = [l for l in project.layerTreeRoot().layerOrder() if l in visible_layers]
         return True
 
+    def renderSingleMetatile(self, metatile):
+        if self.feedback.isCanceled():
+            return
+            # Haven't found a better way to break than to make all the new threads return instantly
+
+        if "Dummy" in threading.current_thread().name: # single thread testing
+            threadSpecificSettings = list(self.settingsDictionary.values())[0]
+        else:
+            try:
+                # guess we're not the only one using TPE in QGIS, cannot assume 0_#, it's sometimes 3 or 4_#
+                threadSpecificSettings = self.settingsDictionary[threading.current_thread().name[-1]] # last number only
+            except:
+                print("Exception! our threads don't match with our settings! ")
```

Comment:
Use the message log or other means for debug messages. Or if the exception really shouldn't happen and you want someone to debug that if it happens, just remove the try-except to raise the exception including stack trace for debugging reasons.